src/main/java/com/es/phoneshop/model/cart/HttpSessionCartService.java

```diff
@@ -32,29 +30,44 @@ public Cart getCart(HttpServletRequest request) {
 
     @Override
     public void add(Cart cart, Product product, int quantity) {
-        Optional<CartItem> cartItem = findItem(cart, product);
-        if (cartItem.isPresent()) {
-            checkEnoughtStock1(quantity,product);
-            checkEnoughtStock2(cartItem,quantity,product);
-            cartItem.get().setQuantity(cartItem.get().getQuantity() + quantity);
-        } else {
-            checkEnoughtStock1(quantity,product);
-            cart.getCartItems().add(new CartItem(product, quantity));
+        if (quantity <= 0) {
```

Comment:
The method is too complicated and full of redundant checks. Simplify it as much as possible